doc/api/test_runner.md

```diff
@@ -0,0 +1,275 @@
+# Test runner
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental
+
+<!-- source_link=lib/test_runner.js -->
+
+The `test_runner` module facilitates the creation of JavaScript tests that
+report results in [TAP][] format. To access it:
+
+```js
+const test = require('test_runner');
+```
+
+Tests created via the `test_runner` module consist of a single function that
+executes either synchronously or asynchronously. Synchronous tests are
+considered passing if they do not throw an exception. Asynchronous tests return
+a `Promise`, and are considered passing if the returned `Promise` does not
+reject. The following example illustrates how tests are written using the
+`test_runner` module.
+
+```js
+const assert = require('assert');
+const test = require('test_runner');
```

Comment:
Any strong reason to use the default export instead of a named export? Typing default exports across CJS / EJS is unnecessarily complex and I'd argue that the ergonomic gains are not worth it. In addition, that leaves the whole export namespace for future improvements and APIs.

Reply:
Consistency with other builtins alone is enough for me to think it worthwhile. I don't think a `default` value could be anything except the whole list due to that precedent. Builtins do provide named exports as well as the default so I'm also unsure about the point about the "whole export namespace". Having `default` also is important for tooling interop that assumes the exports located in `default` position map to the whole list of exports since they assume that about CJS generally.

I'm not clear on the problem being asked if it is about TypeScript or destructuring and actual keyboard issues but I'm unclear on the concrete problem.

Reply:
@bmeck There are not a lot of other builtins whose default export is also a "main" function.

I'd also prefer to have:
```js
const test_runner = require('test_runner');
test_runner.test()

// Or
const { test } = require('test_runner');
test()

// Or
import test_runner from 'test_runner';
test_runner.test()

// Or
import { test } from 'test_runner';
test()
```

Reply:
I wanted to drop the suggestion coming from the challenges of getting the types fixed for the `cluster` module whose exports are difficult to type from an interop perspective: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59058

Reply:
Having the `test` function be the default export also means that all future methods or properties that might be added to `test_runner` need to become properties of the `test` function instead of becoming other first-class, top-level named exports.

Reply:
@targos all four of those are already possible.

Reply:
Exporting just the test function is bound to paint you into a corner.  Export an object of the same type as the argument to the function provided to `test()`, which has a bound `test()` method.